redis_collections/dicts.py

```diff
@@ -75,43 +64,55 @@ def __init__(self, *args, **kwargs):
             As mentioned, :class:`Dict` does not support following
             initialization syntax: ``d = Dict(a=1, b=2)``
         """
+        data = args[0] if args else kwargs.pop('data', None)
         super(Dict, self).__init__(*args, **kwargs)
 
+        if data:
+            self.update(data)
+
+    def _get_hash_dict(self, key, redis):
+        key_hash = hash(key)
+        D = redis.hget(self.key, key_hash)
+        D = {} if D is None else self._unpickle(D)
+
+        return key_hash, D
+
     def __len__(self):
-        """Return the number of items in the dictionary."""
-        return self.redis.hlen(self.key)
+        """Return the number of items in the dictionary.
+
+        .. note::
+            Due to implementation on Redis side, this method is inefficient.
+            The time taken is varies with the number of keys in stored.
```

Comment:
I don't love this, but it's necessary for correctness. Local counting wouldn't catch changes made outside of the application, and counting the number of hashes misses the `hash(x) == hash(y), x != y` case.